src/devices/Bh1745/Bh1745.cs

```diff
@@ -0,0 +1,430 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Drawing;
+using System.Linq;
+
+namespace Iot.Device.Bh1745
+{
+    /// <summary>
+    /// Digital color sensor Bh1745.
+    /// </summary>
+    public class Bh1745 : IDisposable
+    {
+        /// <summary>
+        /// The primary I2c address of the BH1745
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x38;
+
+        /// <summary>
+        /// The secondary I2c address of the BH1745
+        /// </summary>
+        public const byte SecondaryI2cAddress = 0x39;
+
+        /// <summary>
+        /// Gets or sets the state of the software reset.
+        /// True indicates that the initial reset is started, false that it is not started.
+        /// On reset all registers are reset and the sensor goes to power down mode.
+        /// </summary>
+        public bool SoftwareReset
+        {
+            get
+            {
+                var status = Read8BitsFromRegister((byte)Register.SYSTEM_CONTROL);
+                status = (byte)((status & (byte)Mask.SW_RESET) >> 7);
+                return Convert.ToBoolean(status);
+            }
+            set
+            {
+                var status = Read8BitsFromRegister((byte)Register.SYSTEM_CONTROL);
+                status = (byte)((status & (byte)~Mask.SW_RESET) | Convert.ToByte(value) << 7);
+
+                Write8BitsToRegister((byte)Register.SYSTEM_CONTROL, status);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the state of the interrupt pin.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown if invalid InterruptStatus is set.</exception>
+        public InterruptStatus InterruptReset
+        {
+            get
+            {
+                var intReset = Read8BitsFromRegister((byte)Register.SYSTEM_CONTROL);
+                intReset = (byte)((intReset & (byte)Mask.INT_RESET) >> 6);
+                return (InterruptStatus)intReset;
+            }
+            set
+            {
+                if (!Enum.IsDefined(typeof(InterruptStatus), value))
+                    throw new ArgumentOutOfRangeException();
+
+                var intReset = Read8BitsFromRegister((byte)Register.SYSTEM_CONTROL);
+                intReset = (byte)((intReset & (byte)~Mask.INT_RESET) | (byte)value << 6);
+
+                Write8BitsToRegister((byte)Register.SYSTEM_CONTROL, intReset);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the currently set measurement time.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown if invalid MeasurementTime is set.</exception>
+        public MeasurementTime MeasurementTime
+        {
+            get
+            {
+                var time = Read8BitsFromRegister((byte)Register.MODE_CONTROL1);
```

Comment:
Same question for all properties, can these guys be cached? In case not, we should probably change these methods to be a GetMeasurementTime and SetMeasurementTime because of the discussion we are having here: #626